### PR TITLE
Added new a new Github Action to auto-publish components to NPM on tag creation

### PR DESCRIPTION
**Pull request by AaronDDM**
# What did you do?
- [x] Added a new GitHub action to publish components to NPM 
- [x] Added new commands to package.json `yarn release:stable` and `yarn release:canary` to make it easier to publish new versions.
- [x] Added `tsc` for type-checking in our CI flow

# License

I confirm that this contribution is made under the terms of the MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Review by wonka-lighthouse — COMMENTED** [review]
| Device | URL |
|--|--|
| mobile | http://localhost:8000/components/agenda/src/index.html |

     
| Device | URL |
|--|--|
| mobile | http://localhost:8000/components/composer/src/index.html |

Not what you expected? Are your scores flaky? [Run Lighthouse on Foo](https://www.foo.software/lighthouse)
If scores continue to be inconsistent consider [running all audits on Foo](https://github.com/foo-software/lighthouse-check-action#usage-foos-automated-lighthouse-check-api)

**Review by pooja169usp — APPROVED** [review]
Can we please update Readme with the steps?

**Review by heathergh — APPROVED** [review]

**Comment by AaronDDM** [comment]
> Can we please update Readme with the steps?

I'll actually be creating a Guru card for this, since it's an internal process.

File: commons/src/connections/events.ts [hunk 1]
```diff
@@ -64,7 +64,10 @@ export const fetchCalendars = async (
   )
     .then((responses) => {
       const filteredResponses = responses
-        .filter((calendar) => calendar.status === "fulfilled")
+        .filter(
+          (calendar): calendar is PromiseFulfilledResult<Calendar[]> =>
```
Comment (AaronDDM) [inline commons/src/connections/events.ts:68]: Fixed a typing issue.

**Pull request merged**